includes/Checker/Abstract_Check_Runner.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Abstract_Check_runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Checker\Check_Runner;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Preparations\Universal_Runtime_Preparation;
+
+/**
+ * Abstract Check Runner class.
+ *
+ * @since n.e.x.t
+ */
+abstract class Abstract_Check_Runner implements Check_Runner {
+
+	/**
+	 * Instance of the Checks class.
+	 *
+	 * @since n.e.x.t
+	 * @var Checks
+	 */
+	protected $checks;
+
+	/**
+	 * Instance of the Check_Context class.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $context;
+
+	/**
+	 * Array of Check instances to run.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	protected $checks_to_run;
+
+	/**
+	 * Determines if the current request is intended for the plugin checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean Returns true if the check is for plugin else false.
```

Comment:
Per WP docs standards:
```suggestion
	 * @return bool Returns true if the check is for plugin else false.
```